hostinit/setup.go

```diff
@@ -31,93 +31,48 @@ const (
 	scpTimeout = time.Minute
 )
 
-// isHostReady returns whether or not the specified host is ready for its setup script
-// to be run.
-func isHostReady(ctx context.Context, host *host.Host, settings *evergreen.Settings) (bool, error) {
+func setDNSName(ctx context.Context, host *host.Host, settings *evergreen.Settings) error {
 	// fetch the appropriate cloud provider for the host
-	cloudMgr, err := cloud.GetCloudManager(ctx, host.Distro.Provider, settings)
+	cloudMgr, err := cloud.GetManager(ctx, host.Distro.Provider, settings)
 	if err != nil {
-		return false, errors.Wrapf(err, "failed to get cloud manager for provider %s",
-			host.Distro.Provider)
+		return errors.Wrapf(err, "failed to get cloud manager for provider %s", host.Distro.Provider)
 	}
 
-	// ask for the instance's status
-	hostStatus, err := cloudMgr.GetInstanceStatus(ctx, host)
-	if err != nil {
-		return false, errors.Wrapf(err, "error checking instance status of host %s", host.Id)
+	if host.Host != "" {
+		return nil
 	}
 
-	grip.Debug(message.Fields{
-		"message":      "checking host readiness",
-		"runner":       RunnerName,
-		"host_id":      host.Host,
-		"distro":       host.Distro.Id,
-		"id":           host.Id,
-		"local_status": host.Status,
-		"cloud_status": hostStatus,
-	})
-
-	// if the host has failed, terminate it and return that this host is not ready
-	if hostStatus == cloud.StatusFailed {
-		err = errors.WithStack(cloudMgr.TerminateInstance(ctx, host, evergreen.User))
-		if err != nil {
-			return false, err
-		}
-		return false, errors.Errorf("host %s terminated due to failure before setup", host.Id)
+	// get the DNS name for the host
+	hostDNS, err := cloudMgr.GetDNSName(ctx, host)
+	if err != nil {
+		return errors.Wrapf(err, "error checking DNS name for host %s", host.Id)
 	}
 
-	// if the host isn't up yet, we can't do anything
-	if hostStatus != cloud.StatusRunning {
-		return false, nil
+	// sanity check for the host DNS name
+	if hostDNS == "" {
```

Comment:
I think you want to either remove this or change it so we don't get the DNS name unless it isn't already set.